slick/src/main/scala/slick/jdbc/MacroTreeBuilder.scala

```diff
@@ -16,9 +16,23 @@ private[jdbc] class MacroTreeBuilder[C <: Context](val c: C)(paramsList: List[C#
   lazy val rawQueryParts: List[String] = {
     //Deconstruct macro application to determine the passed string and the actual parameters
     val Apply(Select(Apply(_, List(Apply(_, strArg))), _), paramList) = c.macroApplication
-    strArg map {
-      case Literal(Constant(x: String)) => x
-      case _ => abort("The interpolation contained something other than constants...")
+    val list = strArg map { str =>
+      evalString(str).getOrElse(abort("The interpolation contained something other than constants..."))
+    }
+    list.init :+ (list.last + " ")
+  }
+
+  /**
+    * Tries to evaluate the given tree to a String literal known at compile-time
+    */
+  private def evalString(tree: Tree): Option[String] = tree match {
+    case Literal(Constant(x: String)) => Some(x)
+    case Literal(Constant(x)) => Some(String.valueOf(x))
+    case other => try {
+      val x1 = c.untypecheck(q"String.valueOf(${other.duplicate})")
+      Some(c.eval(c.Expr[String](x1)))
+    } catch {
+      case _ : Throwable => None
```

Comment:
Catching Throwable sounds not like a good idea. That includes things where the JVM became instable. Can we narrow it to Exception?